homeassistant/components/switcher_kis/light.py

```diff
@@ -35,16 +35,20 @@ async def async_setup_entry(
     def async_add_light(coordinator: SwitcherDataUpdateCoordinator) -> None:
         """Add light from Switcher device."""
         entities: list[LightEntity] = []
-        if (
-            coordinator.data.device_type.category
-            == DeviceCategory.SINGLE_SHUTTER_DUAL_LIGHT
-        ):
-            entities.extend(SwitcherDualLightEntity(coordinator, i) for i in range(2))
-        if (
-            coordinator.data.device_type.category
-            == DeviceCategory.DUAL_SHUTTER_SINGLE_LIGHT
+        number_of_lights = len(cast(SwitcherLight, coordinator.data).light)
+
+        if coordinator.data.device_type.category in (
+            DeviceCategory.SINGLE_SHUTTER_DUAL_LIGHT,
+            DeviceCategory.DUAL_SHUTTER_SINGLE_LIGHT,
+            DeviceCategory.LIGHT,
         ):
-            entities.append(SwitcherSingleLightEntity(coordinator, 0))
+            if number_of_lights == 1:
```

Comment:
You can do the cast before this line after we know it is a light device

Reply:
yeah it it done.

Reply:
Do you test the PR on a live system? I don't own most of the devices and this PR should show an error in the log before the last commit. 
Can you do a test again on a live system?

Reply:
@thecode I tested with a live system right now. Everything is working as it should, and no errors are found in the logs.

Reply:
@thecode For the record, I tested the previous commit with a live system, and it does result in errors in the logs.
In the recent code, no errors were found.